task/sql-tasks.js

```diff
@@ -280,7 +465,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            Customers.CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice AS "PricePerItem"
+        FROM Customers
+        INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID 
+        INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+        INNER JOIN Products ON OrderDetails.ProductID = Products.ProductID
+        WHERE OrderDetails.UnitPrice =
+        (SELECT
+            MAX(OrderDetails.UnitPrice)
+            FROM Customers AS Customers1
+            INNER JOIN Orders ON Customers1.CustomerID = Orders.CustomerID 
+            INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+            WHERE Customers1.CompanyName = Customers.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.

Reply:
Now the request time is about 39ms (144ms was before)